components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/internal/IdentityRecoveryServiceComponent.java

```diff
@@ -79,11 +82,20 @@ protected void activate(BundleContext bundleContext) {
             bundleContext.registerService(SecurityQuestionPasswordRecoveryManager.class.getName(),
                     new SecurityQuestionPasswordRecoveryManager(JDBCRecoveryDataStore.getInstance(),
                             ChallengeQuestionManager.getInstance()), null);
+            bundleContext.registerService(UserSelfSignUpManager.class.getName(),
+                                          new UserSelfSignUpManager(dataHolder.getRealmService(),
+                                                                    dataHolder.getIdentityEventService()), null);
             bundleContext.registerService(ChallengeQuestionManager.class.getName(),
                     ChallengeQuestionManager.getInstance(), null);
+
+            UserSelfSignUpHandler selfSignUpHandler = new UserSelfSignUpHandler();
+            selfSignUpHandler.setRealmService(dataHolder.getRealmService());
+            selfSignUpHandler.setEventService(dataHolder.getIdentityEventService());
+
+            bundleContext.registerService(AbstractEventHandler.class.getName(), selfSignUpHandler, null);
             IdentityRecoveryServiceDataHolder.getInstance().setRecoveryLinkConfig(new RecoveryLinkConfig());
 
-        } catch (Exception e) {
+        } catch (Throwable e) {
```

Comment:
@kasunbg please comment,  It would be better to throw the exception, rather than catching, so that component is not activated when an error occurred.